src/OrbitQt/orbitmainwindow.cpp

```diff
@@ -1053,13 +1049,13 @@ void OrbitMainWindow::OpenCapture(const std::string& filepath) {
         if (!result.has_value()) {
           QMessageBox::critical(this, "Error while loading capture",
                                 QString::fromStdString(result.error().message()));
-          on_actionEnd_Session_triggered();
+          QApplication::exit(kEndSessionReturnCode);
```

Comment:
Just a personal preference. Could we add a return code like `kEndSessionWithoutWarningReturnCode` for this case? If so, this exit event could be handled in `RequestExitWithReturnCode` with the new return code. WDYT?

Reply:
Vicky has a good point here. Especially since work like `main_thread_executor->AbortWaitingJobs` should be done. Not strictly necessary, but it's probably good to go through `RequestExitWithReturnCode` in all cases.

But I wouldn't use the return code to encode such specific behaviour, since that's an implementation detail of `OrbitMainWindow` that shouldn't leak into other parts of the code. But a second parameter to `RequestExitWithReturnCode` should do the trick.

Reply:
Yes, I agree, the important part is doing the actual clean up like `main_thread_executor->AbortWaitingJobs`. I added another commit, it makes 2 functions out of `RequestExitWithReturnCode`. `ExitWithWarning` and `Exit`. The first one calls the second one, when the warnings are dismissed. The second one cleans up. The second one is called when no warning is needed (When clicking "Cancel" on loading)

ptal